pipelines/package_validate.go

```diff
@@ -378,3 +395,31 @@ func validateRpmUpgrade(ctx context.Context, d *dagger.Client, packages []*dagge
 
 	return nil
 }
+
+// validateSignature uses the given package (rpm) and provided gpg public key to validate signature.
+func validateSignature(ctx context.Context, d *dagger.Client, rpm *dagger.File, packageName string, opts *containers.GPGOpts) error {
+	if ext := filepath.Ext(packageName); ext != ".rpm" {
+		return fmt.Errorf("expected a .rpm file, received '%s'", ext)
+	}
+
+	var (
+		taropts = TarOptsFromFileName(packageName)
+	)
+
+	log.Printf("Validating rpm package signature for v%s%s and platform %s\n", taropts.Version, taropts.Suffix, taropts.Distro)
+
+	code, err := containers.RPMContainer(d, &containers.GPGOpts{Sign: true, GPGPublicKeyBase64: opts.GPGPublicKeyBase64}).
+		WithFile("/src/package.rpm", rpm).
```

Comment:
Would prefer to combine this and the next line in a `/bin/sh -c "..."`

Reply:
You mean the --checksig and the grep?